### ISSUE TITLE
WiiU vWii issue initializing harddrive

### ISSUE THREAD

**Issue by OoIceDeepoO** [Fledge68/WiiFlow_Lite#218]
Premise: WiiFlow has issue mounting hard drives for vWii.

Hi, 
I have read everything I can find and did a ton of tests, so if this is not an issue I am missing something. Here is what is going on. I have been using the latest versions of WiiFlow on my Wii for a couple of years, recently I got a WiiU, modded it with CBHC and began working on getting WiiFlow to work on the vWii. No matter which port I used I could not get WiiFlow to recognize my 4TB fat formatted WD Elements wall powered hard drive, WiiFlow would only boot up (meaning WiiFlow would not hang on splash screen, on a completely clean boot, new install, no save file) if the hard drive was plugged into the bottom USB port. (During all of these tests I only have the Wii hard drive plugged in, not the WiiU one). Though the real problem was after bootup. The SD was always the only option available in game partitions. 

I tried everything: 
1) Turned off SD only mounting. No luck.
2) Tried the top USB port on the back, No luck.
3) Manually edited the .ini file since it would not let me select the USB, no luck.

After all of these tests I was getting sick of the WiiU constanly asking me if I wanted to format the drive every time I booted up so I decided to hide it with UStealth. UStealth had trouble hiding it. Maybe due to the size, since technically you should not be able to format a fat32 4TB Hd, but there are ways... So, I came to the conclusion that it was the hard drive's fault and it was not compatible. 

Anyway, I also had a 2TB Wall Powered Oyen hard drive that I was keeping my WiiU backups on. I decided to switch the two hard drives. The switch went well, the WiiU was fine with the WD Elements harddrive and the Oyen hard drive worked with Ustealth. Again for WiiFlow to even start (not hang on the splash screen), the hard drive had to be plugged into the bottom port. Understand this is just the initial bootup. Using a completely clean 5.4.7 and a deleted WiiFlow save. After the first time boot WiiFlow will load up if the USB is plugged into the top port, but still won't mount the hard drive. Once again I did all my tests no matter what I did, I could not get it to mount the hard drive. 

Finally after alot of testing I found a way for it to recognize the hard drive. If I launch USB Loader GX first, USB Loader GX does it's thing and brings up all the games and can launch them. If then without turning off the WiiU, I go into WiiFlow it finds all the games on the hard drive and will load them. So, USB Loader GX is doing something that is fixing whatever WiiFlow's block is.

Below I have my syscheck and two WiiFlow logs, one where I could not get the USB to mount and one where I loaded USB Loader GX first and then started WiiFlow where the USB mounted. Let me know if you think this is a legitimate bug, if it is not, I am stumped.

**Here is my sys check info:**

SysCheck HDE v2.4.0 HacksDen Edition by JoostinOnline, Double_A, R2-D2199, and Nano
...runs on vIOS58 (rev 6432).

Region: NTSC-U
System Menu 4.3U (v609)

Drive date: 06.28.2011
Homebrew Channel 1.1.2 running on IOS58

Hollywood v0x21
Console ID: 611245917
Console Type: vWii
Shop Channel Country: United States (49)
Boot2 v0
Found 50 titles.
Found 35 IOS on this console. 0 of them are stubs.

vIOS9 (rev 1290): No Patches
vIOS12 (rev 782): No Patches
vIOS13 (rev 1288): No Patches
vIOS14 (rev 1288): No Patches
vIOS15 (rev 1288): No Patches
vIOS17 (rev 1288): No Patches
vIOS21 (rev 1295): No Patches
vIOS22 (rev 1550): No Patches
vIOS28 (rev 2063): No Patches
vIOS31 (rev 3864): No Patches
vIOS33 (rev 3864): No Patches
vIOS34 (rev 3864): No Patches
vIOS35 (rev 3864): No Patches
vIOS36 (rev 3864): No Patches
vIOS37 (rev 5919): No Patches
vIOS38 (rev 4380): No Patches
vIOS41 (rev 3863): No Patches
vIOS43 (rev 3863): No Patches
vIOS45 (rev 3863): No Patches
vIOS46 (rev 3863): No Patches
vIOS48 (rev 4380): No Patches
vIOS53 (rev 5919): No Patches
vIOS55 (rev 5919): No Patches
vIOS56 (rev 5918): No Patches
vIOS57 (rev 6175): No Patches
vIOS58 (rev 6432): USB 2.0
vIOS59 (rev 9249): No Patches
vIOS62 (rev 6942): No Patches
vIOS80 (rev 7200): No Patches
vIOS236 (rev 65535): Trucha Bug, ES Identify, NAND Access
vIOS249[56] (rev 21010, Info: d2x-v10beta52-vWii): Trucha Bug, NAND Access
vIOS250[57] (rev 21010, Info: d2x-v10beta52-vWii): Trucha Bug, NAND Access
vIOS251[58] (rev 21010, Info: d2x-v10beta52-vWii): Trucha Bug, NAND Access
vIOS512 (rev 7): No Patches
vIOS513 (rev 1): No Patches
Report generated on 11/04/2020.

**Here is my WiiFlow log from one of the many times the Hard Drive will not mount:**

Welcome to WiiFlow WFL 5.4.7!
This is the debug output.
WiiU
vWii Mode
AHBPROT disabled = yes
IOS58 v6432
Found WiiFlow Save
Using Port Settings from wiiflow save
AHBPROT disabled after IOS Reload: yes
Now using D2X IOS249[56] v10
FAT Partition at sd: mounted.
Changing USB port to port 0....
Running sound thread
Wiiflow boot.dol Location: sd:/apps/wiiflow
Data Directory: sd:/wiiflow
Wii Games Directory: %s:/wbfs
Nintendont found
GameCube Games Directory: %s:/games
Loading cIOS map
Found d2x base 56 in slot 249
Found d2x base 57 in slot 250
Found d2x base 58 in slot 251
Setting Wii games partition to: 0
emu nand path = sd:/vwiinands/default
saves nand path = sd:/vwiinands/default
Bootup completed!
Creating Gamelist
Adding wii list
Games found: 0
Creating Gamelist
Adding wii list
Games found: 0
Cleanup Coverflow
Stopping sound thread
Stopped sound thread
MEM1_freesize(): 8687600
MEM2_freesize(): 42449536

**Here is my WiiFlow log when I load USB Loader GX first and then WiiFlow:**

Welcome to WiiFlow WFL 5.4.7!
This is the debug output.
WiiU
vWii Mode
AHBPROT disabled = yes
IOS58 v6432
Created WiiFlow Save
AHBPROT disabled after IOS Reload: yes
Now using D2X IOS249[56] v10
FAT Partition at sd: mounted.
Changing USB port to port 0....
 * * * * * * * * * * * *
 * HDD Information
 * Sectors: 3907029167
 * Size [Sector Size]: 1.956 TB [512]
 * * * * * * * * * * * *
FAT Partition at usb1: mounted.
Running sound thread
Wiiflow boot.dol Location: sd:/apps/wiiflow
Data Directory: sd:/wiiflow
Wii Games Directory: %s:/wbfs
Nintendont found
GameCube Games Directory: %s:/games
Loading cIOS map
Found d2x base 56 in slot 249
Found d2x base 57 in slot 250
Found d2x base 58 in slot 251
Setting Wii games partition to: 1
emu nand path = sd:/vwiinands/default
saves nand path = sd:/vwiinands/default
Bootup completed!
Creating Gamelist
Adding wii list
Games found: 71
Displaying covers
Update Play log
Launching game RUUE01
Game ID RUUE01 requested IOS 38.
User selected 0
Changed requested IOS to 249.
Cleanup Coverflow
Stopping sound thread
Stopped sound thread
MEM1_freesize(): 8585200
MEM2_freesize(): 40989056

**Comment by Fledge68** [comment]
try setting force cios off (main setting>startup settings)
I believe usb loader gx defaults to startup with IOS58 which is what happens when you set force cios off.

**Comment by OoIceDeepoO** [comment]
Hi Fledge68,

As always I am a huge fan of your work and think WiiFlow is the best USB loader. Here is the answer to your question and some more info...

My startup settings looks the same regardless of whether I run USB Loader GX first. It actually already had the setting you mentioned above set:
Force Load CIOS = Off
Force CIOS Revision = Auto
USB Port = 0
Mount SD Only = No

The only other clues I could gather when starting WiiFlow are:
If the USB is plugged into the bottom usb port on the back, WiiFlow will load up, but it won't have any games and the USB port is un-selectable in the game partition screen. I only has SD there. (Obviously because the hard drive did not mount. The dump for this is exactly like the first wiiflow.log above that shows the hard drive did not mount.

In contrast when USB Loader GX is loaded while the hard drive is plugged into the top most port on the back of the WiiU, USB Loader GX works great. Then going into WiiFlow it is the exact same as the second dump in my original post, the drive works fine.

I am trying to figure out what USB Loader GX could be doing that would let WiiFlow use the top back port correctly and initialize the hard drive. It does seem like it should be attached to IOS58. The top USB port does not work unless UB Loader GX does something to it first...

Here is what USBLoader GX says when it is booting up:
Loading application cIOS
Loading... Initialize sd card
Loading... Initialize USB  <-- (Something temporary (ends if I turn the WiiU off) is happening either here)
Patching IOS58 <--- (or here)
Checking Installed MIOS...
Loading... Resources

Let me know if there is any other tests I could run.

**Comment by Fledge68** [comment]
well according to your log that you posted you have force cios on.

IOS58 v6432
Found WiiFlow Save
Using Port Settings from wiiflow save
AHBPROT disabled after IOS Reload: yes
Now using D2X IOS249[56] v10

thats why i suggested to turn it off.

if you have force cios on - you could try changing the cios from auto to 250 or 251.

just curious - are you using a Y usb cable to hook your hdd to the wii u?

**Comment by OoIceDeepoO** [comment]
I am using an externally powered USB (plug in, no Y cable). So, I don't think it is an underpowered issue. Just to make sure, I plugged it directly into the wall for tests.

I didn't post the logs from last night, because I went through them line by line and they seemed to match the earlier logs. Do to your comment above, I checked the failure log from last night and the lines match what you copied from my earlier log.
IOS58 v6432
Found WiiFlow Save
AHBPROT disabled after IOS Reload: yes
Now using D2X IOS249[56] v10

From the looks of it, the successful log after USB loader GX looks identical too:
IOS58 v6432
Found WiiFlow Save
AHBPROT disabled after IOS Reload: yes
Now using D2X IOS249[56] v10

Which for both of those I confirmed the settings for startup, but if I understand what you are saying... the logs and what I am saying my settings are, are disagreeing. It should not be switching to IOS249 at this point. It should stay IOS58.  I will see if I can get it to stay. 

I previously tested IOS250, by forcing CIOS and changing Auto to 250, but nothing seemed to work better.

I will do some more tests tonight and see if I can get it to stay on IOS58 in the log and confirm IOS250 does not behave differently.

**Comment by OoIceDeepoO** [comment]
Hi I went through all the tests again. My startup settings have no affect on the logs. Force CIOS is and has been definitely set to no.

1) Top back port will hang on splash screen and have no log. Every time, I cannot even get to the start up settings here, because WiiFlow will not start.  I know what the settings are from the successful loads below.
2) Plugging drive into bottom back port WiiFlow will load up with no games and not mount the hard drive that is plugged into lower port.
3) Top back port will begin to work if and only if USB Loader GX is loaded and exited before it. Every successful log I have where the hard drive mounted is this scenario.

**Comment by OoIceDeepoO** [comment]
HI Follow up questions:
1) Does everyone have this problem or am I the only one? Can other people just launch directly into WiiFlow in the vWii and not have any issues?
2) Any idea of there are any subtle pieces of format the fat32 hard drive that could affect this?

**Comment by OoIceDeepoO** [comment]
Hi Fledge68,

I had some deep thoughts on what could be the issue and just wanted your take. Do you know if other people have the latest version of Wiiflow working fine on a vWii? If so one of the things I am doing is using large hard drives. I have a 4TB and a 2TB. I tried both of them. The large hard drives work fine with WiiFlow on the Wii. The thing about that is that you are not even supposed to be able to format large hard drives in FAT32, but there are programs that let you do it anyway. My thoughts are to do that, maybe these large hard drives have a bad formatting trick that Nintendo fixed in IOS58 for the vWii, but is fine for the Wii. Maybe that is what the "patching IOS58" is doing in USB loader GX. I know there is some sort of boot sector table or something that may or may not be there on large formatted FAT32 drives. I guess my next test would be format a normal reasonable size flash drive with a normal utility like Windows in FAT32 to see how WiiFlow reacts... I will do that and see what happens.

Wayne

**Comment by OoIceDeepoO** [comment]
Hi Fledge68,

I mostly solved this ticket, so you can close it if you want. Here is the story... On the Wii with Wii Flow I am using a 4TB WD easystore drive, that I formatted in FAT32 using a bug in Wii Backup Manager.  Youtube found here https://www.youtube.com/watch?v=Jo613mszHQA

So, I had formatted an externally powered 4TB WD Elements drive using that method to use for the vWii and WiiFlow. That is what was originally not mounting in WiiFlow. I started using a program named Bootice that was mentioned in this article https://gbatemp.net/threads/the-definitive-vwii-hacking-guide.425852/#16 

It still did not mount. I tried switching to a 2TB externally powered Oyen Drive formatted in FAT32 with 32 sectors with Bootice. That too failed.

Which is where you see all our correspondence above. Finally I tried a 16GB Lexar non-powered jump drive formatted with Windows. It booted fine!!! Unfortunately though it mounted and booted every time I tried to launch a game it black screened. I knew the jump drive had an extremely slow read speed. So I ordered a 32GB jump drive from Amazon with a fast read speed formatted it in windows and ... duh duh duh... it worked perfectly!!!

So, bottom line wrapping up. If the read speed is too slow wiiflow will black screen on game launch. Also, there is something wrong with how some of these formatters, format large FAT32 drives. Since technically the supposed limit is 2TB, but most FAT32 formatting programs come no where near to that, there has to be something about the boot sector or table, that IOS58's Wii U version will not play with. USB Loader GX seems to know about this and patches it. Meanwhile on the Wii's version of IOS58 it does not care.

The only curiosity point I have left is what is the biggest drive people have been able to use with Wiiflow on the vWii and how did they format it? If you have answers great, if not I will keep looking. If no comment on that one you can close the ticket. Thanks for the help.

**Comment by OoIceDeepoO** [comment]
I am a fool apparently Fix94 was already aware of all of this, his forwarder patches IOS58 in the same way as USB Loader GX to allow larger FAT32 drives to be used. I was going directly to WiiFlow, no forwarder. See this comment from the old WiiFlow repository at https://code.google.com/archive/p/wiiflow/downloads 

"Wiiflow Forwarder with 3TB+ HDD support, IOS58 and AHBPROT patched out, works in WiiU Wii Mode" I am going to try it with a forwarder tonight and I bet it will allow me to use one of my large hard drives with WiiFlow on the vWii for the first time.

**Comment by AlbertoDiazC** [comment]
Same here. 4TB HDD.
WFL only starts if ulgx is launched beforehand. No other workaround  works for me. Fix94 forwarder doesn't migitate it.

**Comment by OoIceDeepoO** [comment]
Thanks for testing too Alberto, I have been buried with work and haven't had time to work on it in months. It's disappointing, because Wii Flow is easily the best user interface out of all of them, because it is "Game First, System Second". I love the idea of having all the Castlevania's right next to each other and being able to launch any of them regardless of emulator. I wish Wii Flow could be the main interface for even WiiU games, I would settle for it being secondary though and only in the vWii, if I could just unleash it's full power. Hopefully I can test more soon and amass more clues for Fledge.

**Comment by Bastastic** [comment]
I have the same issue? Is there any fix already?

**Comment by Bastastic** [comment]
I also made a gbatemp form about it. https://gbatemp.net/threads/vwii-wiiflow-doesnt-work-with-hdd-but-usb-loader-gx-does.589640/

**Comment by OoIceDeepoO** [comment]
I still have not gotten mine to work. Though I am interested, if anyone else succeeds, I am convinced that IOS58 has to be patched for large hard drives and wii flow just doesn't do it. I would love to be proved wrong though.
